### DISCUSSION TITLE
CSV import can't add records that relate to another table by anything other than id

### DISCUSSION THREAD

**Discussion by sxsws** [directus/directus#13286] [Archive]
### Describe the Bug

e.g. I'm importing a list of countries, which has a relation to continent. In the csv file, if I use the relation value `continent.id` and use the pk values of the continents, the import works fine. However, if I import by `continent.title`, the import fails. I have continent.title set up to not allow dupe values. This seems to cause the fail even though it isn't reported because changing the continent.title to something other than an existing record will allow an import of one record.

I guess that only the .id field value is used as a look up, whereas any other field isn't and is attempted to be inserted instead.

### To Reproduce

try to import data from a csv file where the relation is using another field other than 'someRelation.id`.

### Errors Shown

> This file's data structure doesn't match the collection.

Which I've come to figure means lots of different things. No error shown in console.

### What version of Directus are you using?

9.10.0

### What version of Node.js are you using?

12.22.5

### What database are you using?

mariadb

### What browser are you using?

firefox

### How are you deploying Directus?

local

**Comment by rijkvanzanten** [comment]
That's currently working as expected. The API import expects the same format as the output, so if you provide it a different value than expected in a given column, it will try to save the input as a value for that column. If you send `Rijk` instead of `70642c52-cfdc-4f72-a63f-00a96d54a4f5` for `author`, it will fail as it expected a UUID for the user 👍🏻

**Reply by sxsws** [reply]
ok, that's what I expected, but thought as it was possible to use the key `someRelation.id`, I thought maybe it's possible to use other key names and have it implicitly look them up!
